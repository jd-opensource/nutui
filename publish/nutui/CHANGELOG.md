## v4.0.1

`2023-02-10`

NutUI 非常感谢您对开源事业做出的贡献！🌷🌷🌷
本次社区贡献者 @dongj0316 @mikasayw @rocwong-cn @thx125 @gyt95

* :zap: feat(checkbox): add checkbox new props of shape (#1978) @mikasayw
* :zap: feat(tabs): title bar supports automatic scrolling (#2035) @gyt95
* :zap: feat: calendar 添加自定义方法 (#2084) @lkjh3214
* :zap: feat: 增加 unplugin 插件配置导出 (#2097)  @eiinu
* :zap: feat: 日历组件新增 周选择功能 (#2102)   @lkjh3214
* :zap: feat(input): 新增slots (#2098)  @lkjh3214
* :zap: feat(badge): 新增bubble样式 (#2091) @dongj0316
* :bug: fix(list): list 虚拟列表未设置height，滚动加载有问题 #2088 @szg2008
* :bug: fix(list): 虚拟列表在ios手机上，滚动条滚动到顶部会出现白屏。 #2031 @szg2008
* :bug: fix: barrage，sinature demo修改 @Ymm0008
* :bug: fix(address): 地址问题修改 (#2054)  @yangxiaolu1993
* :bug: fix: drag，configprovide，badge 修改demo @Drjingfubo
* :bug: fix: taro h5 demo修改 @szg2008
* :bug: fix: jdt-style error @Drjingfubo
* :bug: fix: sku组件修改 (#2056)  @yangxiaolu1993
* :bug: fix: collapse,textarea 优化  @Ymm0008
* :bug: fix(tour): demo style edit @Drjingfubo
* :bug: fix(menu): change style @Drjingfubo
* :bug: fix(calendar): demo样式修改，文档修改 (#2052) @lkjh3214
* :bug: fix(config-provider): 覆盖nut-icon属性 (#2022) @thx125
* :bug: fix(table): demo optimization @Drjingfubo
* :bug: fix: imagepreview问题修改 (#2068) @yangxiaolu1993
* :bug: fix: 修复uploader组件的is-deletable在列表模式下不生效 (#2073) @rocwong-cn
* :bug: fix(input): 事件触发修改 (#2081) @lkjh3214
* :bug: fix(notify): resolve teleport warn @Drjingfubo
* :bug: fix(uploader): 修复图片拉伸样式问题 #2026 @richard1015
* :bug: fix: change sideEffects & default export (#2094) @eiinu
* :bug: fix: 公告栏 noticebar 插槽使用时不更新问题修改 (#2104) @yangxiaolu1993
* :bug: fix: 公告栏demo修改 (#2105) @yangxiaolu1993
* :bug: fix: timer ts error resolve @szg2008
* :bug: fix: 签名组件交互优化 @Ymm0008
* 📖 docs(addresslist): taro env use edit  @richard1015
* 📖 docs: popover修改 (#2053) @yangxiaolu1993
* 📖 docs: price 文档问题 (#2064)  @ailululu
* 📖 docs: card edit @Drjingfubo
* 📖 docs(tabbar): taro add punctuation @Drjingfubo
* 📖 docs: 新增4.0介绍 (#2086)  @yangxiaolu1993

## v4.0.0 🚀

`2023-01-12`

### 新特性

- :zap: 全新的升级，基于 vite4 构建
- :sparkles: 轻量化 安装体积由 14MB 下降至 8MB，减少了 39%
- :art: 支持动态定制主题
- :sparkles: 专属图标库`nutui-icons`
- :sparkles: 支持`unplugin-vue-components`插件
- :sparkles: 新增组件`Tour``PullRefresh`

> 完整内容请移步：「[4.0更新介绍](https://nutui.jd.com/next)」

小程序版本依赖 Taro ^3.5.10

NutUI 非常感谢您对开源事业做出的贡献！🌷🌷🌷

本次社区贡献者 @kasonyang @gyt95 @thx125 @Coffee-C @sjx12 @mikasayw @jxt776

### ⚠️ 升级必读

- [从 v3 升级到 v4](https://nutui.jd.com/4x/index.html#/zh-CN/guide/migrate-from-v3)
