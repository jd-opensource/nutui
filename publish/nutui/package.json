{
  "name": "@nutui/nutui",
  "version": "4.0.0-beta.7",
  "description": "京东风格的轻量级移动端 Vue2、Vue3 组件库（支持小程序开发）",
  "main": "dist/nutui.umd.cjs",
  "module": "dist/nutui.es.js",
  "style": "dist/style.css",
  "type": "module",
  "typings": "dist/types/index.d.ts",
  "sideEffects": [
    "dist/packages/_es/*",
    "dist/packages/**/style*"
  ],
  "vetur": {
    "tags": "dist/smartips/tags.json",
    "attributes": "dist/smartips/attributes.json"
  },
  "web-types": "dist/smartips/web-types.json",
  "keywords": [
    "nutui",
    "nutui2",
    "nutui3",
    "nutui4",
    "vue",
    "vue3",
    "webpack",
    "vue component",
    "vue3 component",
    "vite",
    "jdc",
    "jdcfe"
  ],
  "author": "jdcfe",
  "license": "MIT",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org/"
  },
  "scripts": {
    "publish:beta": "npm publish --tag=beta"
  },
  "standard-version": {
    "scripts": {
      "postchangelog": "node ./jd/updateChangelog.js"
    }
  },
  "dependencies": {
    "@nutui/icons-vue": "0.0.19",
    "sass": "^1.50.0",
    "vscode": "^1.1.37",
    "vue-router": "^4.0.12"
  },
  "devDependencies": {
    "@commitlint/cli": "^10.0.0",
    "@commitlint/config-conventional": "^10.0.0",
    "@tarojs/cli": "3.5.6",
    "@tarojs/mini-runner": "3.5.6",
    "@tarojs/plugin-framework-vue3": "3.5.6",
    "@tarojs/plugin-html": "3.5.6",
    "@tarojs/plugin-platform-weapp": "3.5.6",
    "@tarojs/runtime": "3.5.6",
    "@tarojs/taro": "3.5.6",
    "@tarojs/webpack5-runner": "3.5.6",
    "@types/jest": "^26.0.22",
    "@types/node": "^17.0.16",
    "@typescript-eslint/eslint-plugin": "^4.20.0",
    "@typescript-eslint/parser": "^4.20.0",
    "@vitejs/plugin-vue": "^4.0.0",
    "@vue/compiler-dom": "^3.2.45",
    "@vue/compiler-sfc": "^3.2.45",
    "@vue/eslint-config-prettier": "^6.0.0",
    "@vue/eslint-config-typescript": "^7.0.0",
    "@vue/test-utils": "^2.0.0-rc.18",
    "autoprefixer": "^10.3.4",
    "axios": "^0.21.0",
    "babel-preset-taro": "3.5.6",
    "canvas": "^2.9.0",
    "codesandbox": "^2.2.3",
    "eslint": "^7.23.2",
    "eslint-config-taro": "3.5.6",
    "eslint-plugin-prettier": "^3.3.1",
    "eslint-plugin-vue": "^7.8.0",
    "eslint-visitor-keys": "2",
    "fs-extra": "^9.1.0",
    "highlight.js": "^10.3.1",
    "husky": "^6.0.0",
    "inquirer": "^8.2.0",
    "jest": "^26.6.3",
    "lint-staged": "^10.5.0",
    "lzutf8": "0.6.0",
    "markdown-it": "^13.0.1",
    "markdown-it-container": "^3.0.0",
    "prettier": "^2.0.0",
    "remark-codesandbox": "^0.10.1",
    "rollup-plugin-visualizer": "^5.8.3",
    "standard-version": "^9.3.0",
    "swiper": "6.5.1",
    "transliteration": "^2.2.0",
    "ts-jest": "^26.5.5",
    "typescript": "^4.9.3",
    "vite": "^4.0.0",
    "vite-plugin-md": "^0.21.5",
    "vue": "^3.2.24",
    "vue-jest": "^5.0.0-alpha.7",
    "vue-tsc": "^1.0.9",
    "webpack": "^5.69.0"
  },
  "eslintConfig": {
    "root": true,
    "env": {
      "node": true
    },
    "extends": [
      "plugin:vue/vue3-essential",
      "eslint:recommended",
      "@vue/typescript/recommended",
      "@vue/prettier",
      "@vue/prettier/@typescript-eslint"
    ],
    "parserOptions": {
      "ecmaVersion": 2020
    },
    "rules": {
      "vue/no-unused-components": "off",
      "no-debugger": "off"
    }
  },
  "engines": {
    "node": ">= 16.0.0",
    "pnpm": ">= 7.0.0"
  },
  "lint-staged": {
    "*.md": "prettier --write",
    "*.{ts,tsx,js,vue,scss}": "prettier --write"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/jdf2e/nutui.git"
  }
}