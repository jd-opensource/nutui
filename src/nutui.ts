export default {
  async install(app: any): Promise<void> {
    /** webpack */
    // const files = require.context('@/packages', true, /index\.vue$/);
    // files.keys().forEach(component => {
    //   const componentEntity = files(component).default;
    //   app.component(componentEntity.name, componentEntity);
    // });

    /** vite */
    const modules = import.meta.globEager('/src/packages/**/index.vue');
    for (const key in modules) {
      const componentEntity = modules[key]?.default;
      app.component(componentEntity.name, componentEntity);
    }
  }
};
