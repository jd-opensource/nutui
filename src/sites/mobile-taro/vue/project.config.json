{
  "miniprogramRoot": "./dist",
  "projectname": "@nutui/nutui-taro-mobile",
  "description": "nutui-taro-vue",
  "appid": "touristappid",
  "setting": {
    "urlCheck": true,
    "es6": false,
    "postcss": false,
    "minified": false
  },
  "compileType": "miniprogram"
}
