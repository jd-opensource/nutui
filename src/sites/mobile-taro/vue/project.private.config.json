{
  "setting": {},
  "condition": {
    "plugin": {
      "list": []
    },
    "game": {
      "list": []
    },
    "gamePlugin": {
      "list": []
    },
    "miniprogram": {
      "list": [
        {
          "name": "business/pages/address/index",
          "pathName": "business/pages/address/index",
          "query": "",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/noticebar/index",
          "pathName": "feedback/pages/noticebar/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/progress/index",
          "pathName": "feedback/pages/progress/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/swiper/index",
          "pathName": "feedback/pages/swiper/index",
          "query": "",
          "scene": null
        }
      ]
    }
  }
}