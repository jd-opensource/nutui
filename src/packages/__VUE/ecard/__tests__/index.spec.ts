import { DOMWrapper, mount } from '@vue/test-utils'
import { Ecard } from '@nutui/nutui'
import { nextTick } from 'vue'

const data = [
  {
    price: 10
  },
  {
    price: 20
  },
  {
    price: 30
  },
  {
    price: 40
  }
]
test('should render ecard', async () => {
  const wrapper = mount(Ecard, {
    props: {
      dataList: data
    }
  })
  expect(wrapper.html()).toMatchSnapshot()
})
test('should render correct money', async () => {
  const wrapper = mount(Ecard, {
    props: {
      dataList: data,
      modelValue: 0
    }
  })
  const item = wrapper.findAll('.nut-ecard__list__item')

  item[0].trigger('click')
  await nextTick()
  expect((wrapper.emitted('update:modelValue') as any)[0][0]).toBe(10)
})

test('input change when more than maxValue', async () => {
  const wrapper = mount(Ecard, {
    props: {
      dataList: data,
      modelValue: 0,
      cardAmountMax: 100
    }
  })
  const input: DOMWrapper<Element> = wrapper.find('.nut-ecard__list__input--input')
  ;(input.element as HTMLInputElement).value = '123'
  input.trigger('input')
  await nextTick()
  expect((wrapper.emitted('update:modelValue') as any)[0][0]).toEqual(100)
})

test('input change when less than maxValue', async () => {
  const wrapper = mount(Ecard, {
    props: {
      dataList: data,
      modelValue: 0,
      cardAmountMin: 100
    }
  })
  const input: DOMWrapper<Element> = wrapper.find('.nut-ecard__list__input--input')
  ;(input.element as HTMLInputElement).value = '90'
  input.trigger('input')
  await nextTick()
  expect((wrapper.emitted('update:modelValue') as any)[0][0]).toEqual(100)
})
test('input change when less than maxValue', async () => {
  const wrapper = mount(Ecard, {
    props: {
      dataList: data,
      modelValue: 10,
      cardAmountMin: 100
    }
  })
  const input: DOMWrapper<Element> = wrapper.find('.nut-ecard__list__input')
  const add = wrapper.find('.nut-icon')
  input.trigger('click')
  add.trigger('click')
  await nextTick()
  expect((wrapper.emitted('update:modelValue') as any)[0][0]).toBe('')
})
