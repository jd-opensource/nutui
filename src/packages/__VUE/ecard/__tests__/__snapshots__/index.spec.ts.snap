// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render ecard 1`] = `
"<view class="nut-ecard">
  <view class="nut-ecard__title">请选择电子卡面值</view>
  <view class="nut-ecard__list">
    <view class="nut-ecard__list__item">10</view>
    <view class="nut-ecard__list__item">20</view>
    <view class="nut-ecard__list__item">30</view>
    <view class="nut-ecard__list__item">40</view>
    <view class="nut-ecard__list__input">
      <view>其他面值</view>
      <view class="nut-ecard__list__input--con"><input class="nut-ecard__list__input--input" type="text" placeholder="请输入1-5000整数"> ¥</view>
    </view>
    <view class="nut-ecard__list__step">
      <view>¥0</view>
      <view class="nut-input-number">
        <view class="nut-input-number__icon nut-input-number__left nut-input-number__icon--disabled"><svg class="nut-icon nut-icon-minus" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" role="presentation">
            <path d="M299.52 460.8h409.6c28.16 0 51.2 23.04 51.2 51.2s-23.04 51.2-51.2 51.2h-409.6c-28.16 0-51.2-23.04-51.2-51.2s23.04-51.2 51.2-51.2z" fill="currentColor" fill-opacity="0.9"></path>
          </svg></view><input type="number" min="1" max="9999">
        <view class="nut-input-number__icon nut-input-number__right"><svg class="nut-icon nut-icon-plus" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" role="presentation">
            <path d="M345.6 460.8h409.6q51.2 0 51.2 51.2t-51.2 51.2H345.6q-51.2 0-51.2-51.2t51.2-51.2zM550.4 256q51.2 0 51.2 51.2v409.6q0 51.2-51.2 51.2t-51.2-51.2V307.2q0-51.2 51.2-51.2z" fill="currentColor" fill-opacity="0.9"></path>
          </svg></view>
      </view>
    </view>
  </view>
</view>"
`;
